src/body.rs

```diff
@@ -10,7 +13,7 @@ pub struct Body {
 
 impl Body {
     /// Instantiate a `Body` from a reader.
-    pub fn new<R: Read + 'static>(reader: R) -> Body {
+    pub fn new<R: Read + Seek + 'static>(reader: R) -> Body {
```

Comment:
They wouldn't be identical `Body` structs.

Again, looking into other libraries, it doesn't look like it's easy to always do this. For case where the body is string, it is handled, and reqwest can do so also. But for streams, there isn't a way to  reset the stream, and so the 307 response is returned directly. I don't know how I'd feel about `Reusable` and `Oneshot`; it'd all depend on if it was explicit, and not confusing.

Since everyone usually always checks the status code of a response, it should seem like people would notice when they receive a 307.